plugins/in_splunk/splunk_prot.c

```diff
@@ -546,22 +545,17 @@ static int process_hec_raw_payload(struct flb_splunk *ctx, struct splunk_conn *c
                                    struct mk_http_request *request)
 {
     int ret = -1;
-    int type = -1;
     struct mk_http_header *header;
 
     header = &session->parser.headers[MK_HEADER_CONTENT_TYPE];
     if (header->key.data == NULL) {
         send_response(conn, 400, "error: header 'Content-Type' is not set\n");
         return -1;
     }
-    else if (header->val.len == 10 &&
-        strncasecmp(header->val.data, "text/plain", 10) == 0) {
-        type = HTTP_CONTENT_TEXT;
-    }
-    else {
+    else if (header->val.len != 10 ||
```

Comment:
@cosmo0920 please check this changes (the operator changed from `&&` to `||`

Reply:
That's fine. This is because these lines of representation was created from: 
```c
!(header->val.len == 10 && strncasecmp(header->val.data, "text/plain", 10) == 0
```
Apply the negate operators (== -> != and && -> ||):
```c
header->val.len != 10 || strncasecmp(header->val.data, "text/plain", 10) != 0
```